pandas/core/extensions.py

```diff
@@ -0,0 +1,233 @@
+        # type: () -> ExtensionDtype
+        pass
+
+    @property
+    def shape(self):
+        # type: () -> T.Tuple[int, ...]
+        return (len(self),)
+
+    @property
+    def ndim(self):
+        # type: () -> int
+        """Extension Arrays are only allowed to be 1-dimensional"""
+        return 1
+
+    @property
+    @abc.abstractmethod
+    def nbytes(self):
+        # type: () -> int
+        # TODO: default impl?
+        pass
+
+    # ------------------------------------------------------------------------
+    # Additional Methods
+    # ------------------------------------------------------------------------
+    @abc.abstractmethod
+    def isna(self):
+        # type: () -> T.Sequence[bool]
+        # TODO: narrow this type?
+        pass
+
+    # ------------------------------------------------------------------------
+    # Indexing methods
+    # ------------------------------------------------------------------------
+    @abc.abstractmethod
+    def take(self, indexer, allow_fill=True, fill_value=None):
+        # type: (T.Sequence, bool, T.Optional[T.Any]) -> ExtensionArray
+        """For slicing"""
+
+    @abc.abstractmethod
+    def take_nd(self, indexer, allow_fill=True, fill_value=None):
+        """For slicing"""
+        # TODO: this isn't nescesary if we only allow 1D (though maybe
+        # impelment it).
+
+    @abc.abstractmethod
+    def copy(self, deep=False):
+        # type: (bool) -> ExtensionArray
+        """For slicing"""
+
+    # ------------------------------------------------------------------------
+    # Block-related methods
+    # ------------------------------------------------------------------------
+    @property
+    def fill_value(self):
+        # type: () -> T.Any
+        # TODO
+        return None
+
+    @abc.abstractmethod
+    def formatting_values(self):
+        # type: () -> np.ndarray
+        # At the moment, this has to be an array since we use result.dtype
+        """An array of values to be printed in, e.g. the Series repr"""
+
+    @classmethod
+    @abc.abstractmethod
+    def concat_same_type(cls, to_concat):
+        # type: (T.Sequence[ExtensionArray]) -> ExtensionArray
+        """Concatenate multiple array
+
+        Parameters
+        ----------
+        to_concat : sequence of this type
+
+        Returns
+        -------
+        cls
+        """
+
+    @abc.abstractmethod
+    def get_values(self):
+        # type: () -> np.ndarray
+        # TODO: What is the required return value? Sequence? ndarray?, ...?
+        # Categorical does an ndarray
+        """Get the underlying values backing your data
+        """
+        pass
+
+    @property
+    @abc.abstractmethod
+    def can_hold_na(self):
+        # type: () -> bool
+        pass
+
+    @property
+    def is_sparse(self):
+        # type: () -> bool
+        return False
+
+    def slice(self, slicer):
+        # TODO: is this right?
+        # In general, no. Probably just remove it?
```

Comment:
Currenlty in geopandas, I needed a custom slicer on the Block level (so I think we need at least something here or on the block level)

Reply:
Sorry, by "remove" I meant remove the default implementation and make it abstract.